### DISCUSSION TITLE
Lesson 4 Challenge

### DISCUSSION THREAD

**Discussion by Brandoneos** [smartcontractkit/full-blockchain-solidity-course-py#1610] [Q&A] [answered]
I'm around 4 hours in the python course and I keep getting these "Could not find files for the given pattern(s)." which I searched up and said it doesn't mean anything is wrong but I also get a Traceback call every time I run my deploy.py script. It shows a bunch of directories in my files and I really don't want these things to pop in every time I run, I just want to see the printed statements(I'm on windows if this does anything). Does anyone have a solution or am I just messing up the code?

![pythonerror2](https://user-images.githubusercontent.com/89663097/181382510-cfa9afbf-549c-4bf0-acef-e6fc8184d16a.jpg)

**Comment by cromewar** [comment, marked as answer]
Hello @Brandoneos 
As you know `Could not find files for the given pattern(s).` is no an error, however the rest of the log you are receiving is not related to this, you are getting the cytools error, the solution is here:  https://medium.com/@cromewar/the-mighty-cytools-error-at-web3-smart-contract-development-with-brownie-63335d50f230

Cheers.

**Reply by Brandoneos** [reply]
Hmm.. I tried installing Cython and having XCODE. I did not try the first option though. How exactly would I download a previous version of web3? I searched on google and am having trouble. Thanks for your response!

**Reply by cromewar** [reply]
you can specify the version when you install the package, as you are using a mac, I strongly suggest to:
1. setup a python virtualenv with version 3.8
2. install py-solc-x on that version

does this help?

**Reply by Brandoneos** [reply]
Not really. I have no idea how to setup a virtual env or if it will help. I don't have a ton of experience with python. I don't know if the Javascript version of this course is more updated for these version mismatches, but for now I will just skip over my errors and keep learning through the course. Thanks though, and I will look into what is a virtual env in the future.

**Reply by cromewar** [reply]
Actually this errors are more related with windows itself than the packages... let's just say windows is a mess managing all this libraries that's why I always recommend using WSL.
Anyways the packages for python is `virtualenv` this will allow you to create an isolated python instance for your projects.
Cheers

**Reply by Brandoneos** [reply]
Okay I see how to install WSL now from the beginning of the course in github, do you recommend that I setup this WSL before installing any web3.py, pip3, and other packages on my computer? Is it best if I delete everything and install WSL after downloading VScode but before installing any packages?

**Reply by cromewar** [reply]
WSL is and isolated operating system inside windows so you can have both instances with separated libraries. So re-installing everything in WSL should not affect your windows install.

**Reply by Brandoneos** [reply]
Thanks so much for answering my questions!
